cmd/router/main.go

```diff
@@ -37,32 +38,30 @@ import (
 
 var log = logf.Log.WithName("InferenceGraphRouter")
 
-func callService(serviceUrl string, input []byte, res chan<- []byte) error {
+func callService(serviceUrl string, input []byte) ([]byte, error) {
 	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
 	if err != nil {
-		log.Error(err, "An error has occurred for service %s", serviceUrl)
-		return err
+		log.Error(err, "An error has occurred from service", "service", serviceUrl)
+		return nil, err
 	}
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		log.Error(err, "error while reading the response")
-		return err
 	}
-	res <- body
-	return nil
+	return body, err
 }
 
 func pickupRoute(routes []v1alpha1.InferenceStep) *v1alpha1.InferenceStep {
```

Comment:
@yuzisun : Can you explain how this function is working ? How is it taking care of weighted distribution ?